lib/foreman_maintain/concerns/base_database.rb

```diff
@@ -140,6 +140,38 @@ def raise_psql_missing_error
                 ' Make sure system has psql utility installed.'
       end
 
+      # rubocop:disable Metrics/MethodLength
+      def amcheck
+        # creating the extension and executing the check requires superuser privileges
+        return unless local?
+
+        psqlcmd = "runuser - postgres -c 'psql --set=ON_ERROR_STOP=on #{configuration['database']}'"
+
+        amcheck_create = 'CREATE EXTENSION IF NOT EXISTS amcheck;'
+
+        execute!(psqlcmd, :stdin => amcheck_create)
```

Comment:
@ehelms would you prefer if this would be a `return unless amcheck_installed?` and I add "install amcheck" to foreman/pulpcore/candlepin puppet?